internal_run_task.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+// A private struct we need for unmarshalling
+type internalRunTask struct {
+	ID          string                 `jsonapi:"primary,tasks"`
+	Name        string                 `jsonapi:"attr,name"`
+	URL         string                 `jsonapi:"attr,url"`
+	Description string                 `jsonapi:"attr,description"`
+	Category    string                 `jsonapi:"attr,category"`
+	HMACKey     *string                `jsonapi:"attr,hmac-key,omitempty"`
+	Enabled     bool                   `jsonapi:"attr,enabled"`
+	RawGlobal   map[string]interface{} `jsonapi:"attr,global-configuration,omitempty"`
```

Comment:
TIL that jsonapi doesn't automatically decode json annotated attribute object types. It's certainly in spec to have an object within an attribute. I will take a closer look after I finish this review.

Reply:
Unfortunately, jsonapi is effectively a dead project :-( It had a "No Maintenance Intended" badge added 4 years ago and the PR/Issues list is wasteland. We'd have to fork it if we wanted to fix any issues with it.